cassovary-core/src/main/scala/com/twitter/cassovary/algorithms/centrality/AbstractCentrality.scala

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2014 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.twitter.cassovary.algorithms.centrality
+
+import com.twitter.cassovary.graph.DirectedGraph
+
+abstract class AbstractCentrality(graph: DirectedGraph) extends Centrality {
+  val centrality = new Array[Double](graph.maxNodeId + 1)
+
+  /**
+   * Normalize the values
+   * @return
+   */
+  def normalize: Array[Double] = centrality.map(c => c / (graph.maxNodeId - 1))
```

Comment:
It seems that nodeCount returns the number of "active" nodes in the graph (i.e. nodes that are connected to other nodes).  However, I would argue that dividing by (maxNodeId - 1) is the right call here.

Lets consider a graph of 15 nodes with a fully connected subgraph of 5 nodes, therefore there are ten nodes in the graph that are not connected.  The method nodeCount would tell me that there are 5 nodes in the graph (which is correct IF I only want to consider active connections), however, the method maxNodeId would tell me that there are 15 nodes in the graph.  So now the question becomes how do I "normalize" the in-degree and out-degree centrality?  I would argue that the "best" normalization would be to divide by the maximum number of *possible* connections per node (which would be 14) *not* the maximum number of possible connections among currently active nodes.  

Thoughts?